iterators_task_1/student/include/impl.h

```diff
@@ -10,4 +12,32 @@
  *
  * @warning YOU MUST USE STRING STREAMS AND STREAM ITERATORS!
  */
-std::string reverseWords(const std::string& input);
+struct line : public std::string {};
+std::istream & operator >> (std::istream& stream, line& token)
+{
+    std::getline(stream,token,' ');
```

Comment:
1. That is not line reading
2. By default, stream delimiter is a space. So you don't need that at all